Number_types/include/CGAL/Exact_integer.h

```diff
@@ -50,30 +50,22 @@ typedef unspecified_type Exact_integer;
 
 #else // not DOXYGEN_RUNNING
 
-#if CGAL_USE_GMPXX
-
+#ifdef CGAL_USE_GMPXX
 typedef mpz_class Exact_integer;
-
-#elif CGAL_USE_GMP
-# ifdef CGAL_USE_BOOST_MP
-typedef boost::multiprecision::mpz_int Exact_integer;
-# else
+#elif defined(CGAL_USE_GMP)
+#if defined(CGAL_USE_BOOST_MP)
+typedef BOOST_gmp_arithmetic_kernel::Integer Exact_integer;
+#else
 typedef Gmpz Exact_integer;
-# endif
-
-#elif CGAL_USE_LEDA
-
+#endif
+#elif defined(CGAL_USE_LEDA)
 typedef leda_integer Exact_integer;
-
-#elif CGAL_USE_CORE
-
+#elif defined(CGAL_USE_BOOST_MP)
+typedef BOOST_cpp_arithmetic_kernel::Integer Exact_integer;
+#elif defined(CGAL_USE_CORE)
 typedef CORE::BigInt Exact_integer;
```

Comment:
As someone mentioned recently, this case is probably dead code (cannot use Core without GMP), but cleaning that up is unrelated to this patch. I wonder if we should add `#else #error "cannot determine a bigint type"` or if continuing is on purpose, in case we don't actually use Exact_integer later.

Reply:
I have added `#else #error ....`.